jsonnet/thanos-mixin/alerts/sidecar.libsonnet

```diff
@@ -10,9 +10,8 @@
               message: 'All Thanos Sidecars are unhealthy for {{ $value }} seconds.',
             },
             expr: |||
-              count(time() - max(thanos_sidecar_last_heartbeat_success_time_seconds{%(thanosSidecarSelector)s}) by (pod) > 300) == 0
+              absent(time() - max(thanos_sidecar_last_heartbeat_success_time_seconds{%(thanosSidecarSelector)s}) by (pod) > 300)
```

Comment:
This will only alert if the entire metric is absent/unavailable. So it will only fire if we don't have any `thanos_sidecar_last_heartbeat_success_time_seconds{%(thanosSidecarSelector)s}` metrics. I think you're looking for something else.

Reply:
Oww, I tried with a similar metric and it kind of worked. So do you think the previous `count` approach is better?

Reply:
You were right as always `absent` wouldn't work as I've anticipated. I have fixed it now.